grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy

```diff
@@ -166,8 +166,24 @@ class ProductAvailabilityService {
             }
             picklistItems {
                 if (product) {
-                    requisitionItem {
-                        eq("product", product)
+                    or {
+                        requisitionItem {
+                            eq("product", product)
+                        }
+                        orderItem {
+                            eq("product", product)
+                        }
+                    }
+                }
+                requisitionItem {
+                    isNotNull("id")
+                }
+                inventoryItem {
+                    ne("lotStatus", LotStatusCode.RECALLED)
+                }
+                binLocation {
+                    locationType {
```

Comment:
I think this is a bit more tricky than that, because `HOLD_LOCATION` is `supportedActivity` instead of type (take a look at `Location.isOnHold()`